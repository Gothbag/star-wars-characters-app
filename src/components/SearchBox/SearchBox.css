.search-box {
	margin-bottom: 5px;
}

.search-box span {
	
}

.search-box .btn {
	height: 35px;
	width: 33%;
	height: 100%;
	padding: .3rem .8rem .7rem;
}

.search-box input {
	margin-right: 1%;
	width: 65%;
}