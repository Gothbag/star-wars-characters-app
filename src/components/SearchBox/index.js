import React from "react";
import PropTypes from "prop-types";
import {noop} from "lodash";

import "./SearchBox.css"

class SearchBox extends React.PureComponent {

    constructor(props){
        super(props)
        this.state = {
            value: ""
        }
    }

    static propTypes = {
        handleOnClick: PropTypes.func
    }

    static defaultProps = {
        handleOnClick: noop
    }

    handleOnSearch = () => {
        const { state: { value } } = this;
        const trimmedVal = value.trim();
        if (trimmedVal === "") {
            return;
        }
        this.props.handleOnSearch(trimmedVal);
    }

    handleOnChange = event => {
        const { target: { value } } = event;
        this.setState({
            value
        });
    } 

    handleOnKeyPress = event => {
        if(event.key === "Enter"){
            this.handleOnSearch();
        }
    }
    
    render() {
        const { handleOnChange, handleOnKeyPress, handleOnSearch, state: { value } } = this;
        return (<div className="input-group search-box">
            <input type = "text" value={value} onChange={handleOnChange} onKeyPress={handleOnKeyPress}/>
            <button className="btn" type="button" onClick={handleOnSearch}>
                Search
            </button>
        </div>);
    }
}

export default SearchBox;
