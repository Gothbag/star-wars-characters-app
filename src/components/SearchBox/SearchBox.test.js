import React from 'react';
import SearchBox from './index';

describe("<SearchBox/>", () => {

	const props = {

	};

	beforeEach(() => {
		props.handleOnSearch = jest.fn();
	});

	it('renders without crashing', () => {
		const wrapper = shallow(<SearchBox/>);
		expect(wrapper.find("button").length).toBe(1);
	});

	it("should call the handleOnSearch prop upon clicking the search button if it's not empty", () => {
		const wrapper = shallow(<SearchBox {...props}/>);
		wrapper.find("input").simulate("change", {target:{value:"ok"}});
		wrapper.find("button").simulate("click");
		expect(props.handleOnSearch.mock.calls.length).toBe(1);
	});

	it("shouldn't call the handleOnSearch prop upon clicking the search button if it's empty", () => {
		const wrapper = shallow(<SearchBox {...props}/>);
		wrapper.find("button").simulate("click");
		expect(props.handleOnSearch.mock.calls.length).toBe(0);
	});

	it("should call the handleOnSearch prop upon pressing Enter on the input if it's not empty", () => {
		const wrapper = shallow(<SearchBox {...props}/>);
		wrapper.find("input").simulate("change", {target:{value:"er"}});
		wrapper.find("input").simulate("keypress", {key:"Enter"});
		expect(props.handleOnSearch.mock.calls.length).toBe(1);
	});

});
