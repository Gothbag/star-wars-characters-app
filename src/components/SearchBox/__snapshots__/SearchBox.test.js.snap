// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox/> renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox
    handleOnClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
/>,
        <button
          className="btn"
          onClick={[Function]}
          type="button"
>
          Search
</button>,
      ],
      "className": "input-group search-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "onKeyPress": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search",
          "className": "btn",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Search",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value=""
/>,
          <button
            className="btn"
            onClick={[Function]}
            type="button"
>
            Search
</button>,
        ],
        "className": "input-group search-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "onKeyPress": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
            "className": "btn",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
